gateway/src/apicast/policy/ip_check/client_ip.lua

```diff
@@ -23,7 +24,13 @@ local function ip_from_x_forwarded_for_header()
     return nil
   end
 
-  return re.split(forwarded_for, ',', 'oj')[1]
+  -- THREESCALE-5258 forwarded_for can contain port. If port is in there IP
+  -- value will not be parsed correctly, parsing as url will get the correct
+  -- host
+  -- `:` split is not valid in case that it's IPv6 notation.
+  local host = re.split(forwarded_for, ',', 'oj')[1]
+  local uri = resty_url.parse("my://".. host)
```

Comment:
Why is `my://` needed here?

Reply:
if no scheme, the resty_url will fail, so it's the safest way to detect if IP/PORT is in there, also with IPv4/IPv6 combitation.

Reply:
:+1:
It'd be useful to add a comment that clarifies that.

Reply:
It's already in there :\

Reply:
ah sorry, I missed that.